### DISCUSSION TITLE
Need help understanding load-image

### DISCUSSION THREAD

**Discussion by lc9er** [janet-lang/janet#1274] [Q&A] [answered]
I'm trying to understand how to load an image saved to disk, in order to restore a repl session. (load-image (slurp "someFile.jimage")) returns a table...and I'm not sure what to do after that. I'm not able to grasp how to access the data loaded, if this can be used to restore a new repl to the saved state in what was loaded, or if I've just misunderstood the use of load-image.

**Comment by CosmicToast** [comment, marked as answer]
Images operate on environment tables (just like `import`), so you need to merge it in.
Example to save a repl:
```janet
(def a :a)
(curenv) # demo to show :a got defined
(spit "env" (make-image (curenv)))
```
Demo to load the state:
```janet
# merge with no prefix like with "use", do not re-export
(merge-module (curenv) (load-image (slurp "env")) "" false)
a # => :a
```

**Reply by lc9er** [reply]
Got it and I see what I was missing (merge-module). Thank you for taking the time to demo this.

**Comment by ianthehenry** [comment]
`use` and `import` can load images in addition to source files, so if you want to restore an image in `someFile.jimage`, you can call `(use ./someFile)`. And if you're developing with the built-in repl, you can run `janet -l ./someFile` to start the repl with your previous image loaded into the initial environment.
